kong/router/atc.lua

```diff
@@ -322,25 +321,45 @@ end
 -- split port in host, ignore form '[...]'
 -- example.com:123 => example.com, 123
 -- example.*:123 => example.*, 123
-local function split_host_port(h)
-  if not h then
-    return nil, nil
-  end
+local split_host_port
+do
+  local tonumber = tonumber
+  local DEFAULT_HOSTS_LRUCACHE_SIZE = DEFAULT_MATCH_LRUCACHE_SIZE
 
-  local p = h:find(":", nil, true)
-  if not p then
-    return h, nil
-  end
+  local memo_hp = lrucache.new(DEFAULT_HOSTS_LRUCACHE_SIZE)
+
+  split_host_port = function(h)
```

Comment:
```suggestion
  split_host_port = function(key)
```

Doesn't seems that we need another variable of `key` below.

Reply:
done.